### PR TITLE
Adding script to validate hook_update for field storage changes

### PR DESCRIPTION
**Pull request by piggito**
When the site is live we should provide hook_update() whenever we perform changes in field storage configs.

We need a new sniffer to validate that hook_update() is provided along with config change

### CODE REVIEW

**Comment by piggito** [comment]
Still need to consider removal of field.storage config objects and make CI job manual with a comment about the need to make it required for live projects

**Comment by piggito** [comment]
Added change to make CI job manual and also confirmed that we don't need hook_update when field.storage config is removed so just need review to merge.

**Comment by andypost** [comment]
Nice trick! needs someone to test on fresh install)

File: scripts/makefile/field-storage-hook-update.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (piggito) [reply scripts/makefile/field-storage-hook-update.sh]: Fixed please recheck

File: scripts/makefile/tests.mk [hunk 2]
```diff
@@ -58,6 +58,15 @@ compval:
 	@echo "Composer.json validation..."
 	@docker run --rm -v `pwd`:`pwd` -w `pwd` $(IMAGE_PHP) composer validate --strict
 
+## Validate hook_update_N() for field storage changes
+hookupdateval:
+ifneq ("$(wildcard scripts/makefile/field-storage-hook-update.sh)","")
```
[the remark this answers is not in the document]
Comment (piggito) [reply scripts/makefile/tests.mk]: Fixed please recheck

File: scripts/makefile/field-storage-hook-update.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+FIELD_STORAGE_COUNT=$(git show --name-status | grep "^M" | grep field.storage | wc -l)
+
+if [ "$FIELD_STORAGE_COUNT" -gt "0" ]; then
+  # If storage changed we need new hook_updates in install file.
+  UPDATE_COUNT=$(git show web/modules/custom/*.install | grep "^\+" | grep _update_ | wc -l)
+  if [ "$UPDATE_COUNT" -gt "0" ]; then
+    printf "hook_update_N() included for storage changes."
+    exit 0
+  else
+    printf "hook_update_N() missing for storage changes."
```
[the remark this answers is not in the document]
Comment (piggito) [reply scripts/makefile/field-storage-hook-update.sh]: Fixed please recheck

**Pull request merged**